test/device/test_device_configuration.py

```diff
@@ -142,6 +142,23 @@ def test_can_select_device_loaded_from_config_file(moler_config, device_factory)
     assert device.__class__.__name__ == 'UnixLocal'
 
 
+def test_can_clone_device(moler_config, device_factory):
+    conn_config = os.path.join(os.path.dirname(__file__), os.pardir, "resources", "device_config.yml")
+    moler_config.load_config(config=conn_config, config_type='yaml')
+
+    device_org = device_factory.get_device(name='UNIX_LOCAL')
+    assert device_org is not None
+    device_cloned = device_factory.get_cloned_device(source_device=device_org, new_name="CLONED_UNIX_LOCAL")
```

Comment:
Why we need `source_device` as parameter? Notice that `device_factory.get_device()` besides returning device is caching it. Wouldn't it be simpler just to call:
```python
device_cloned = device_factory.get_cloned_device(source_device_name="UNIX_LOCAL", new_name="CLONED_UNIX_LOCAL")
```

Reply:
Good point.